src/seedu/addressbook/Main.java

```diff
@@ -78,16 +79,33 @@ private void exit() {
 
     /** Reads the user command and executes it, until the user issues the exit command.  */
     private void runCommandLoopUntilExitCommand() {
-        Command command;
+        Command command = null;
         do {
-            String userCommandText = ui.getUserCommand();
-            command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
-            recordResult(result);
-            ui.showResultToUser(result);
-
+        	try {
+                String userCommandText = ui.getUserCommand();
+                checkStorageFileExistence();
+                command = new Parser().parseCommand(userCommandText);
+                CommandResult result = executeCommand(command);
+                recordResult(result);
+                ui.showResultToUser(result);		
+        	} catch (FileDeletedException e) {
+        		System.out.println(e.getMessage());
+        		System.exit(1);
+        	}
         } while (!ExitCommand.isExit(command));
     }
+    
+    /**
+     * Checks whether directory file was deleted during execution
+     * @throws FileDeletedException
+     */
+    private void checkStorageFileExistence() throws FileDeletedException {
```

Comment:
Good idea to extract this code into a method.

However, shouldnt this method return a true/false rather than using exception? i.e. let the caller decide what to do (including throwing exception). Exception indicates something outside of the normal execution scenario, in this case, this method _is_ supposed to check for existence of a file and not finding the file is a common outcome.